implementation.js

```diff
@@ -149,28 +161,14 @@ module.exports = function from(arrayLike) {
 		}
 	}
 
-	var len = ES.ToLength(items.length);
+	var arrayLike = ES.ToObject(parseIterableLike(items));
+	var len = ES.ToLength(arrayLike.length);
 	var A = ES.IsCallable(C) ? ES.ToObject(new C(len)) : new Array(len);
```

Comment:
Ah yes, additionally, `mapFn` also has a similar problem. But I want that those problems are fixed in another pr because this change has no regression.